### PR TITLE
69915 unit tests for ReaderServicesManager

### PR DESCRIPTION
**Pull request by dawidkeshop**

### CODE REVIEW

File: src/Tests/CaptainHook.Tests/Builders/SubscriberConfigurationBuilder.cs [hunk 1]
```diff
@@ -53,7 +53,7 @@ public SubscriberConfigurationBuilder WithOidcAuthentication()
             return this;
         }
 
-        public SubscriberConfigurationBuilder WithCallback(string uri)
+        public SubscriberConfigurationBuilder WithCallback(string uri = "https://calback.eshopworld.com")
```
Comment (tabman83) [inline src/Tests/CaptainHook.Tests/Builders/SubscriberConfigurationBuilder.cs:56]: Typo

File: src/Tests/CaptainHook.Tests/Director/ReaderServicesManagerTests.cs [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+                "fabric:/CaptainHook/EventReader.testevent-subscriber1-10000000000000",
+                "fabric:/CaptainHook/EventReader.testevent.completed-captain-hook-10000000000000");
+
+            VerifyFabricClientDeleteCalls();
+
+            VerifyServiceCreatedEventPublished("fabric:/CaptainHook/EventReader.testevent-captain-hook-10000000000000",
+               "fabric:/CaptainHook/EventReader.testevent-subscriber1-10000000000000",
+               "fabric:/CaptainHook/EventReader.testevent.completed-captain-hook-10000000000000");
+
+            VerifyServiceDeletedEventPublished();
```
Comment (tabman83) [inline src/Tests/CaptainHook.Tests/Director/ReaderServicesManagerTests.cs:70]: You would probably need an AssertionScope() here

File: src/Tests/CaptainHook.Tests/Builders/SubscriberConfigurationBuilder.cs (hunk 1 above)
Comment (dawidkeshop) [reply src/Tests/CaptainHook.Tests/Builders/SubscriberConfigurationBuilder.cs:56]: Good catch, thanks!

File: src/Tests/CaptainHook.Tests/Director/ReaderServicesManagerTests.cs (hunk 2 above)
Comment (dawidkeshop) [reply src/Tests/CaptainHook.Tests/Director/ReaderServicesManagerTests.cs:70]: Great idea, thanks!

File: src/Tests/CaptainHook.Tests/Director/ReaderServicesManagerTests.cs [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+        {
+            _dateTimeProviderMock.SetupGet(x => x.UtcNow)
+                .Returns(DateTimeOffset.MinValue.AddMilliseconds(10000000000000));
+            var generator = new ReaderServiceNameGenerator(_dateTimeProviderMock.Object);
+            return new ReaderServicesManager(_fabricClientMock.Object, _bigBrotherMock.Object, _dateTimeProviderMock.Object, generator);
+        }
+
+        private void ProgressTime(int milliseconds)
+        {
+            _dateTimeProviderMock.SetupGet(x => x.UtcNow).Returns(DateTimeOffset.MinValue.AddMilliseconds(10000000000000 + milliseconds));
```
Comment (chomick) [inline src/Tests/CaptainHook.Tests/Director/ReaderServicesManagerTests.cs:49]: You could simply 'get' the current one and set the expectation with increased value - no need to hardcode the base then

File: src/Tests/CaptainHook.Tests/Director/ReaderServicesManagerTests.cs [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+
+        private readonly Mock<IBigBrother> _bigBrotherMock = new Mock<IBigBrother>();
+        private readonly Mock<IFabricClientWrapper> _fabricClientMock = new Mock<IFabricClientWrapper>();
+        private readonly Mock<IDateTimeProvider> _dateTimeProviderMock = new Mock<IDateTimeProvider>();
+
+        private ReaderServicesManager CreateReaderServiceManager()
+        {
+            _dateTimeProviderMock.SetupGet(x => x.UtcNow)
+                .Returns(DateTimeOffset.MinValue.AddMilliseconds(10000000000000));
+            var generator = new ReaderServiceNameGenerator(_dateTimeProviderMock.Object);
```
Comment (chomick) [inline src/Tests/CaptainHook.Tests/Director/ReaderServicesManagerTests.cs:43]: We could mock that as well maybe?

File: src/CaptainHook.DirectorService/Infrastructure/ReaderServiceNameGenerator.cs [hunk 5]
```diff
@@ -26,7 +26,7 @@ public string GenerateNewName(SubscriberNaming naming)
             return $"{readerServiceNameUri}-{newSuffix}";
         }
 
-        public IList<string> FindOldNames(SubscriberNaming naming, IList<string> serviceList)
+        public IList<string> FindOldNames(SubscriberNaming naming, IEnumerable<string> serviceList)
```
Comment (chomick) [inline src/CaptainHook.DirectorService/Infrastructure/ReaderServiceNameGenerator.cs:29]: Do we have unit tests for that class and its two methods?

Comment (dawidkeshop) [reply src/CaptainHook.DirectorService/Infrastructure/ReaderServiceNameGenerator.cs:29]: Yes we do, both methods was well covered as part of Nino's task. 
This change is only result of refactoring, we never need any IList features for that list, so I decided to change it everywhere to IEnumerable.

File: src/Tests/CaptainHook.Tests/Director/ReaderServicesManagerTests.cs (hunk 3 above)
Comment (dawidkeshop) [reply src/Tests/CaptainHook.Tests/Director/ReaderServicesManagerTests.cs]: You're right, thanks!

**Pull request merged**